packages/api/src/external/carequality/document/shared.ts

```diff
@@ -0,0 +1,95 @@
+import {
+  DocumentReference,
+  OutboundDocumentQueryResp,
+  OutboundDocumentRetrievalResp,
+  DocumentReference as IHEGWDocumentReference,
+} from "@metriport/ihe-gateway-sdk";
+import {
+  DocumentReference as DocumentReferenceFHIR,
+  DocumentReferenceContent,
+} from "@medplum/fhirtypes";
+import { toFHIRSubject } from "@metriport/core/external/fhir/patient/index";
+import { cqExtension } from "../../carequality/extension";
+import { DocumentReferenceWithId, createDocReferenceContent } from "../../fhir/document";
+import { metriportDataSourceExtension } from "../../fhir/shared/extensions/metriport";
+
+export type DocumentWithMetriportId = DocumentReference & {
+  id: string;
+  originalId: string;
+};
+
+type IHEResults = OutboundDocumentQueryResp | OutboundDocumentRetrievalResp;
+
+// Create a single array of all the document references from all the document query results
+export function combineDocRefs(documentQueryResults: IHEResults[]): DocumentReference[] {
```

Comment:
That doesnt work though. Each IHEResult has an array of doc references which i reduce so that i can map each results documentReferences, add the url and then add it to an overall array.

Reply:
It would be a `.flatMap` instead of `.map` then. But NABD.

Also, when adding a doc to a function, its better to do so as TSDoc so its visible when the caller hovers over the function.

Reply:
Smart will do that